source/common/http/utility.cc

```diff
@@ -768,6 +768,14 @@ const std::string& Utility::getProtocolString(const Protocol protocol) {
   NOT_REACHED_GCOVR_EXCL_LINE;
 }
 
+absl::string_view Utility::getScheme(const RequestHeaderMap& headers) {
+  if (Runtime::runtimeFeatureEnabled("envoy.reloadable_features.use_scheme_header") &&
+      !headers.getSchemeValue().empty()) {
+    return headers.getSchemeValue();
+  }
+  return headers.getXForwardedProtoValue();
```

Comment:
What are the cases in which scheme is not but XFP is set? I thought we should always set scheme now early on in the request lifecycle if it's not otherwise set? Can you add more comments?